contracts/mocks/TimelockReentrant.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../utils/Address.sol";
+
+contract TimelockReentrant {
```

Comment:
This contract is not timelock specific. IMO it should have a generic name like "ReentrancyTester"

Also, it looks like its "just" a contract that exposes a `run` (the one you call `reenter`) function which itself does an arbitrary call. Unless you put the address of this particular contract in the `target` ... this doesn't actually reenter at all.

IMO we should find a better name / design for this helper contract.

Reply:
I don't think it'll work. If we generalize this contract we'll end up with logic shared across tests without context. Similarly to [this previous comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4177#pullrequestreview-1405421223), it may become difficult to parse a generic "Reentrant" contract considering that we may need different behaviors depending on the case.

For example, sometimes we may need to emit an event in the reentrancy to check it (reentrancy may not fail as in ERC4626), or we may need to catch a specific error (eg. using `Address.functionCall(...)` with fallback error), or maybe checking a value that was changed.

In this particular contract, I used `_reenterData` because I saw the comment about generalizing coming. I'd be okay with any of the following options:

1. Keep it as it is rn
2. Instead of `_reenterData`, use the Timelock interface to make a call directly to both `execute` or `executeBatch`
3.  Making a `Reentrant.mock.sol` with kinds of reentrancy in which the current `TimelockReentrant` will become one of them

Reply:
ok, lets merge that, and I'll propose a more generic helper in another PR at some point.

Reply:
> Instead of _reenterData, use the Timelock interface to make a call directly to both execute or executeBatch
You'll have a fixpoint problem if you try to do that without an helper (the proposal data needs to contain itself)